app/main/presenters/search_presenters.py

```diff
@@ -288,8 +288,18 @@ def _annotate_categories_with_selection(lot_slug, category_filters, request, url
             content_manifest, framework, aggregations, keys_to_remove, parent_category=category['value'])
 
         if selected_descendants:
-            # If a parentCategory has been sent as a url query param, de-select all other parent categories.
-            if request.values.get('parentCategory', category['value']) != category['value']:
+            # If a parentCategory has been sent as a url query param, and it's this category...
+            # (Note: `get` fallback is for if there are selected descendants but no parent category is set, which is
+            # not expected, but could happen if constructing URLs by hand.)
+            if request.values.get('parentCategory', category['value']) == category['value']:
+                # ... then ensure this choice survives as a hidden field for the filters form
+                parent_category_filter = dict()
+                parent_category_filter['name'] = 'parentCategory'
+                parent_category_filter['value'] = category['value']
```

Comment:
I might be missing something obvious here but is there any reason why this isn't written on one line as 
```
parent_category_filter = {'name': 'parentCategory', 'value': category['value']}
```
?

Reply:
Point taken, this was just done in the same style as where we build other "filter dictionaries" - see `search_presenters.build_lots_and_categories_link_tree` for example. Some of those cases it's perhaps more justifiable to use this format as some of the values are the result of relatively long function calls (e.g. see the `root_node['link'] = search_link_builder(...)` line).

I think if I change this I should change them all, and I'm not sure the end result would necessarily be more readable?

Reply:
Or even just

```
selected_category_filters.append({
    'name': 'parentCategory',
    'value': category['value'],
})
```

Reply:
Hmm. Consistency probably wins in this case? Your call @galund...

Reply:
Going with consistency for now if that's ok. I'll bear in mind that ppl prefer `{..}` notation next time I'm writing this kind of thing :)